### ISSUE TITLE
Attention mask type must be padding or padding_causal for qkv_format=thd!

### ISSUE THREAD

**Issue by 1049451037** [NVIDIA/TransformerEngine#1005]
Why must I use `padding_causal`? Can't I just use `causal`?

**Comment by cyanguwa** [comment]
Hi @1049451037 , the `padding_causal` here for THD is equivalent to the `causal` for BSHD or SBHD. The 'padding' keyword is required because THD naturally implies padding (the sequences are of different lengths) and our backends have to take that into account when doing softmax. Thanks.

**Comment by 1049451037** [comment]
However, from my point of view (Megatron view), there should me no difference between `padding_causal` and `causal`. Because now that we have `causal`, previous tokens cannot even see the tokens backward. So it doesn't matter whether there are paddings in the end.

It's a bit confusing for me.

**Comment by guyueh1** [comment]
Is there a fix to make program not crash when I am using THD data and `attn_mask=="causal"` ?
Or is it that for such case, an error is supposed to be thrown
